design/one-pager-template-stacks.md

```diff
@@ -0,0 +1,344 @@
+
+The TSM should make the `Stack` the [Owner Reference](https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/#owners-and-dependents) for any managed instances of the managed kind (`Wordpress`). This will ensure that when the Stack is removed, all managed resources created from that Stack are also removed.
+
+### RBAC
+
+The Stack Manager currently creates a Service Account for Stack Deployments, restricted by the Owned and Dependent types declared by the Stack. A Template Stack Manager will benefit from this same processing with little or no changes needed. The roles for each TSM controller will be tailored to the types that Stack needs to handle and render. The TSM will also need a `Role` permitting it to read the `Stack` it is operating on. It is important for each TS to take advantage of the Stack `spec.dependsOn` field to define all of the resources defined in the template.
+
+An alternative to deploying a TSM per Stack would be to have the existing Stack Manager handle all new TSM defined types. The SM runs with cluster privileges, giving Template Stacks to much control. Under this approach, the existing Stack Manager could still create Service Accounts and take advantage of User Impersonation when reconciling for types defined within each Template Stack. This approach would not offer simple visibility into the roles the deployment is actively using.
+
+By creating a TSM deployment for each Template Stack that is installed, we are able to scope the permissions given to each TSM instance to the permissions requested by the Stack (i.e. the CRDs it owns and depends on). This is analogous to how the Stack Manager currently creates a `Deployment` to host each controller-runtime based Stack’s controller which is also scoped down to the needed permissions.
+
+#### Concerning the use of Core Types
+
+In several examples throughout this design, Template Stacks are demonstrated generating `Deployment` resources by including them in template text. There are compelling justifications for both permitting and denying Template Stacks from declaring a dependency on core and extension types, such as `deployment.apps/v1`.
+
+The ability to use Template Stacks to render arbitrary Kubernetes resources is a powerful feature. Such a privilege would change the boundaries of a Stack, permitting it to function outside of Crossplane managed types. Template Stacks would compete with `KubernetesApplication` on this capability.
+
+If Template Stacks are prohibited from using core and extension types, the complement of `KubernetesApplication` and a local `KubernetesCluster` can be used to overcome the limitation while providing additional security boundaries. These are features and challenges to investigate in a future design.
+
+### Life Cycle and Updates
```

Comment:
Below questions are not specific to template stacks and valid for current Stacks as well, right?

Reply:
I've added more text to this section and tried to split the thinking up into `Resource life cycle` and `Stack life cycle`.  I believe I may have omitted some useful details from the `Stack life cycle` section that may or may not have been mentioned elsewhere in the design.

Let me know if you find more holes.